src/main/java/de/zalando/aruha/nakadi/repository/zookeeper/ZookeeperUtils.java

```diff
@@ -0,0 +1,28 @@
+package de.zalando.aruha.nakadi.repository.zookeeper;
+
+import org.apache.curator.framework.recipes.locks.InterProcessLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+public class ZookeeperUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZookeeperUtils.class);
+
+    private ZookeeperUtils() {
+    }
+
+    public static <V> V runLocked(final Callable<V> callable, final InterProcessLock lock) throws Exception {
+        try {
+            lock.acquire();
+            return callable.call();
```

Comment:
If the exception is thrown I don't have guarantee that it was not acquired. So stay on the safe side it makes sense for me to try to release it.